ext/src/http/client/curl/http_client_curl.cc

```diff
@@ -416,6 +420,23 @@ void HttpClient::MaybeSpawnBackgroundThread()
             still_running = 1;
           }
 
+          if (still_running > 0)
+          {
+            continue;
+          }
+
+          // If there is no pending jobs, Exit flush thread if there is not data to flush more
+          // than one minute.
+          if (self->background_thread_wait_for_ != std::chrono::milliseconds{0})
```

Comment:
We should avoid delays after calling `Shutdown`, or graceful exiting will take too long.